source/features/link-to-changelog-file.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import {BookIcon} from '@primer/octicons-react';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import {buildRepoURL, getRepo} from '../github-helpers';
+
+const getCacheKey = (): string => `changelog:${getRepo()!.nameWithOwner}`;
+
+function parseFromDom(): string | false {
+	return select('.js-navigation-item [title^="changelog." i]')?.textContent ?? false;
+}
+
+async function fetchFromApi(): Promise<string | false > {
+	const {repository} = await api.v4(`
+		repository() {
+			object(expression: "HEAD:") {
```

Comment:
There's a `defaultBranchRef` I think

Reply:
```graphql
{
  repository(owner: "sindresorhus", name: "refined-github") {
    defaultBranchRef {
      target {
        ... on Tree {
          entries {
            name
          }
        }
      }
    }
  }
}

```

Cant get this to work

Reply:
Refer to other features, we use it elsewhere

Reply:
I know. But its not working (its an issue on github's end)

Reply:
https://github.community/t/on-tree-fails-on-defaultbranchref/161041

Reply:
It confused me a bit too, but this made it clear:

```gql
defaultBranchRef {
	target {
		__typename
	}
}
```

The type of `target` is `Commit`, so `Tree` just does not match. To get the tree from `Commit` you use:

```gql
{
  repository(owner: "sindresorhus", name: "refined-github") {
    defaultBranchRef {
      target {
        ... on Commit {
          tree {
            entries {
              name
            }
          }
        }
      }
    }
  }
}
```

But probably there's an even shorter way, avoiding `defaultBranchRef.target` altogether. 🤔

Reply:
I'm not sure my example is worth it given the length.

Reply:
Should we get the default branch `await defaultBranch` so we dont have issues with branches/N/slashes?

Reply:
This wouldn’t be any better than just using HEAD: